### DISCUSSION TITLE
Can't get the middleware example to work

### DISCUSSION THREAD

**Discussion by beowulf1416** [actix/actix-web#2400] [Q&A] [answered]
I copied the example from the page https://actix.rs/docs/middleware/ about creating a middleware but I cannot get it to work. The ```Service``` trait requires 1 generics argument but 2 generics argument is being passed.

**Comment by robjtede** [comment, marked as answer]
I suspect you're using the v4 beta which uses a slightly tweaked Service trait. See docs for it here: https://docs.rs/actix-service/2.0.1/actix_service/trait.Service.html. Also read the build-in middleware for some extra guidance: https://github.com/actix/actix-web/tree/master/src/middleware.
